.github/workflows/clang-tidy.yml

```diff
@@ -0,0 +1,36 @@
+name: clang-tidy
+
+on:
+  push: {}
+
+jobs:
+  job:
+    name: run-clang-tidy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: install clang-tidy
+      run: sudo apt update && sudo apt -y install clang-tidy
+
+    - name: create build environment
+      run: cmake -E make_directory ${{ runner.workspace }}/_build
+
+    - name: configure cmake
```

Comment:
do you need cmake to run clang-tidy?

Reply:
Indirectly because clang-tidy uses compile_commands.json and I generate it using this cmake command.

Reply:
ah makes sense. thanks.